Makefile

```diff
@@ -347,6 +350,44 @@ bundle-run: operator-sdk ## Run bundle image
 bundle-cleanup: operator-sdk ## Remove bundle installed via bundle-run
 	$(OPERATOR_SDK) -n $(BUNDLE_RUN_NAMESPACE) cleanup $(OPERATOR_NAME)
 
+# Build a file-based catalog image
+# https://docs.openshift.com/container-platform/4.14/operators/admin/olm-managing-custom-catalogs.html#olm-managing-custom-catalogs-fb
+# NOTE: CATALOG_DIR and CATALOG_DOCKERFILE items won't be deleted in case of recipe's failure
+CATALOG_DIR := catalog_dir
+CATALOG_DOCKERFILE := ${CATALOG_DIR}.Dockerfile
+CATALOG_INDEX := $(CATALOG_DIR)/index.yaml
+
+.PHONY: catalog-build
+catalog-build: opm ## Build a file-based catalog image.
+	# Remove the catalog directory and Dockerfile if they exist
+	-rm -r ${CATALOG_DIR} ${CATALOG_DOCKERFILE}
+	@mkdir -p ${CATALOG_DIR}
+	$(OPM) generate dockerfile ${CATALOG_DIR}
+	$(OPM) init ${OPERATOR_NAME} \
+		--default-channel=${CHANNELS} \
+		--description=./README.md \
+		--icon=./config/assets/medik8s_blue_icon.png \
+		--output yaml \
+		> ${CATALOG_INDEX}
+	$(OPM) render ${BUNDLE_IMG} --output yaml >> ${CATALOG_INDEX}
+	$(MAKE) add_channel_entry_for_the_bundle
+	$(OPM) validate ${CATALOG_DIR}
+	docker build . -f ${CATALOG_DOCKERFILE} -t ${CATALOG_IMG}
+	# Clean up the catalog directory and Dockerfile
+	rm -r ${CATALOG_DIR} ${CATALOG_DOCKERFILE}
```

Comment:
Why would you need to delete it twice? Deletion, in the beginning, makes more sense unless there is a problem of having the index image exist.
FYI we can modify https://github.com/medik8s/machine-deletion-remediation/blob/main/.gitignore to not include this file so it won't be stored upstream if that's the issue

Reply:
Yes I understand we can move it under `.gitignore`, but I don't like to have dangling files/folders :grin: 

The need for deleting it twice is because I couldn't find a nicer way to clean up these items _in case the target fails_.
So, if the target succeeds, it correctly cleans them up at the end, however it also needs to wipe out any leftovers from any previous unsuccessful run (if any).

Reply:
I think having the first deletion is important for a fresh start.
>but I don't like to have dangling files/folders 😁

Ok, your call. I don't see an issue with that and we already do similar stuff. For example, the binaries that we use (operator-sdk, opm, etc. under the `bin` directory) are also locally created and stored...

Reply:
> the binaries...

which is fine as we reuse them and there is no need to download them over and over again, while there is no point storing these items